### PR TITLE
Migrated the project to the Go modules for dependency management

### PR DESCRIPTION
**Pull request by ZapAnton**
An attempt at and closes https://github.com/exercism/cli/issues/535

### CODE REVIEW

**Comment by ZapAnton** [comment]
Since appveyor has passed all the test, I assume this PR is ready for the review.
Let me know if additional steps are needed

**Comment by ZapAnton** [comment]
@kytrinyx , done

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,33 +10,28 @@ Exercism would be impossible without people like you being willing to spend time
...
+1. `git checkout -b <development-branch-name>`
+1. `git push -u origin <development-branch-name>` (setup where you push to, check it works)
+
+Then make your desired changes and submit a pull request. Please provide tests for the changes where possible.
 
-Then make changes as usual and submit a pull request. Please provide tests for the changes where possible.
+Please note that if your development directory is located inside the `GOPATH`, you would need to set the `GO111MODULE=on` environment variable, in order to be able to use the `modules` system. 
 
-If you care about the details, check out the blog post [Contributing to Open Source Repositories in Go][contrib-blog] on the Splice blog.
+If you wish to learn how to contribute to the Go projects without the `modules`, check out the blog post [Contributing to Open Source Repositories in Go][contrib-blog] on the Splice blog.
```
Comment (kytrinyx) [inline CONTRIBUTING.md]: Let's simplify this. Now that the workflow is the normal GitHub workflow, I think it's enough to say:

> Fork this repository, and clone your fork.
> Please note that if your development directory is located inside the `GOPATH`, you would need to set the `GO111MODULE=on` environment variable, in order to be able to use the `modules` system.

The splice blog post is no longer relevant to this.

Comment (kytrinyx) [reply CONTRIBUTING.md]: Actually, let me pull this in and update the docs myself. I tested that this works :-)

**Pull request merged**